base-test/src/main/java/io/fabric8/launcher/base/test/hoverfly/LauncherHoverflyEnvironment.java

```diff
@@ -0,0 +1,92 @@
+package io.fabric8.launcher.base.test.hoverfly;
+
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+
+import io.fabric8.launcher.base.Paths;
+import org.junit.contrib.java.lang.system.ProvideSystemProperty;
+
+import static io.fabric8.launcher.base.test.hoverfly.LauncherHoverflyRuleConfigurer.isHoverflyInSimulationMode;
+
+/**
+ * Gives a way:
+ * - to have fixed environment only when Hoverfly is in simulation mode.
+ * - to use the Launcher truststore
+ */
+public class LauncherHoverflyEnvironment extends ProvideSystemProperty {
+
+
+    private final String host;
+    private final String port;
+    private final boolean simulationMode;
+    private File trustStoreTempFolder;
+
+    private LauncherHoverflyEnvironment(String host, String port) {
+        this.host = host;
+        this.port = port;
+        this.simulationMode = isHoverflyInSimulationMode();
+    }
+
+    public static LauncherHoverflyEnvironment createHoverflyEnvironment(String host, String port) {
+        return new LauncherHoverflyEnvironment(host, port);
+    }
+
+    public static LauncherHoverflyEnvironment createDefaultHoverflyEnvironment() {
+        return new LauncherHoverflyEnvironment("127.0.0.1", "8558");
+    }
+
+    @Override
+    protected void before() throws Throwable {
+        initTrustStore();
+        and("https.proxyHost", host);
+        and("https.proxyPort", port);
+        and("javax.net.ssl.trustStore", trustStoreTempFolder.getAbsolutePath() + "/hoverfly.jks");
+        and("javax.net.ssl.trustStorePassword", "changeit");
+
+        super.before();
+    }
+
+    @Override
+    protected void after() {
+        super.after();
+        deleteTrusStoreTempFolder();
+    }
+
+    public LauncherHoverflyEnvironment andForSimulationOnly(final String name, final String value) {
+        if (simulationMode) {
+            super.and(name, value);
+        }
+        return this;
+    }
+
+    @Override
+    public LauncherHoverflyEnvironment and(final String name, final String value) {
+        super.and(name, value);
+        return this;
+    }
+
+    private File createTemporaryFolderIn(File parentFolder) throws IOException {
+        File createdFolder = File.createTempFile("junit-hoverfly", "", parentFolder);
+        createdFolder.delete();
+        createdFolder.mkdir();
+        return createdFolder;
+    }
+
+    private void initTrustStore() throws IOException {
+        trustStoreTempFolder = createTemporaryFolderIn(null);
```

Comment:
I think it's just a file, why do you need a temp folder?

Reply:
to use copy :) which is simpler

Reply:
But if you insist :p I would just create a file (after seeing your perf concerns :).

Reply:
You can copy directly to the temporary folder afaik, `Paths.get(System.getProperty("java.tmp.dir"))` or something like that will give you the tmp folder path

Reply:
but Files.createTempFile is easier, I believe :smile:

Reply:
My last update is doing that :) I ended up with the same conclusion..